packages/block-brokers/src/trustless-gateway/utils.ts

```diff
@@ -55,3 +55,71 @@ export async function * findHttpGatewayProviders (cid: CID, routing: Routing, lo
     yield new TrustlessGateway(uri, { logger, transformRequestInit: options.transformRequestInit })
   }
 }
+
+/**
+ * A function that handles ensuring the content-length header and the response body is less than a given byte limit.
+ *
+ * If the response contains a content-length header greater than the limit or the actual bytes returned are greater than
+ * the limit, an error is thrown.
+ */
+export async function limitedResponse (response: Response, { log, byteLimit, signal }: { log?: Logger, byteLimit: number, signal?: AbortSignal }): Promise<Uint8Array> {
```

Comment:
If `byteLimit` is not optional, let's not pass it as an option:

```suggestion
export async function limitedResponse (response: Response, byteLimit: number, options?: AbortOptions): Promise<Uint8Array> {
```

Also, I think we can lose the logger here - the caller can log any thrown errors.